rust/worker/src/system/wrapped_message.rs

```diff
@@ -1,6 +1,30 @@
 use super::{Component, ComponentContext, Handler};
 use async_trait::async_trait;
 use std::fmt::Debug;
+use tokio::sync::oneshot;
+
+// Why is this separate from the WrappedMessage struct? The size of WrappedMessage
+// must be constant for all messages on a given component, so the alternative would
+// be to add a field to WrappedMessage like Box<dyn ReplyableMessage<C>>
+// (similar to how the outgoing message is boxed). Unfortunately this doesn't work
+// as the message generic (M) would have to be declared at the method level and this
+// is incompatible with dynamic dispatch.
+// (https://doc.rust-lang.org/error_codes/E0038.html#method-has-generic-type-parameters)
+#[derive(Debug)]
+pub(crate) struct MessageWithReplyChannel<M: Debug + Send + 'static, Result: Send> {
```

Comment:
nit - could we call this HandleableMessageImpl since that is really all its purpose is - i think that naming may reduce the cognitive overhead here a bit